### PR TITLE
Only use ES version 2 in docker setup

### PR DESCRIPTION
**Pull request by czue**
## Summary

image

I wasn't totally sure how far down this road to go. I removed the ES7 docker container and config entries. I also removed the test target. I stopped short of doing any broader updates to the codebase and also did not rename the `elasticsearch2` container to `elasticsearch` (since I thought it might make for a more complicated upgrade/migration experience).

One thing I'm not sure of is how to support anyone currently doing dev on ES7 easily? Or whether we just don't worry about those people and they should switch back to 2?

Casting a wide net with review to get others to weigh in.

## Safety Assurance

- [x] Risk label is set correctly
- [x] All migrations are backwards compatible and won't block deploy
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change
- [x] If QA is part of the safety story, the "Awaiting QA" label is used
- [x] I have confidence that this PR will not introduce a regression for the reasons below

Dev environments only. I tested this by symlinking `localsettings.example.py` to `localsettings.py` and doing a clean install from scratch. Both normal `runserver` and docker `runserver` seemed to work correctly.

### Automated test coverage

The tests use docker so hopefully they'll catch any problems.

### QA Plan

N/A

### Safety story

Dev only.

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### CODE REVIEW

**Review by millerdev — APPROVED** [review]
Thanks for untangling all these details.

**Comment by orangejenny** [comment]
Fantastic.

> One thing I'm not sure of is how to support anyone currently doing dev on ES7 easily? Or whether we just don't worry about those people and they should switch back to 2?

Is there any advantage to using ES7 locally? I think we should just remove 7 until we're ready to upgrade production, for the sake of simplicity.

**Review by snopoke — APPROVED** [review]

**Comment by czue** [comment]
> Is there any advantage to using ES7 locally? I think we should just remove 7 until we're ready to upgrade production, for the sake of simplicity.

Cool this sounds good to me. I just always worry about introducing breaking changes.

**Comment by calellowitz** [comment]
> > Is there any advantage to using ES7 locally? I think we should just remove 7 until we're ready to upgrade production, for the sake of simplicity.
> 
> Cool this sounds good to me. I just always worry about introducing breaking changes.

I think the risk of someone using ES7 locally introducing a breaking change, due to incompatibilities is worth the inconvenience of forcing people back to ES2 (and I say that as someone who uses ES7 because I assumed the container called `elasticsearch` was supposed to be the default). This is especially true since most developer's local data is very small and therefore easy to reindex. As long as we communicate that clearly, I think it is fine to do, unless you also have concerns about external users with these containers in production.

**Review by calellowitz — APPROVED** [review]

**Comment by biyeun** [comment]
excellent! I think as we do upgrades (like the upcoming ES3 > ES5) the default container should not be switched over to the new version until we are confident it won't break production

**Comment by czue** [comment]
@dimagi/team-commcare-hq this changes the default set up for dev environments using Docker. In particular, ElasticSearch 7 is now no longer supported and the container is no longer built / run. And the ElasticSearch 2 container is now running on port 9200 (previously 6200).

It is recommended that anyone who was on ES7 switch back to ES2. If you were using ES7 (check `settings.ELASTICSEARCH_MAJOR_VERSION`) then you should change `ELASTICSEARCH_MAJOR_VERSION` to 2 and rebuild your elasticsearch databases (`./manage.py ptop_preindex`) against ES2.

If you were already using ES2, then you may need to change `ELASTICSEARCH_PORT` to `9200` (from `6200`) if you were using Docker.

Feel free to ping me if you have any issues after this.

File: docker/hq-compose-services.yml [hunk 1]
```diff
@@ -56,7 +45,7 @@ services:
       ES_CLUSTER_NAME: ES_CLUSTER_NAME
       ES_JAVA_OPTS: "-Des.script.engine.groovy.inline.aggs=true -Des.script.engine.groovy.inline.search=true"
     ports:
-      - "6200:6200" # is used in python-elasticsearch-v7 build matrix
+      - "9200:9200" # is used in python-elasticsearch-v7 build matrix
```
Comment (proteusvacuum) [inline docker/hq-compose-services.yml:48]: looks like this comment is not valid any more?

Comment (czue) [reply docker/hq-compose-services.yml:48]: oops, good catch, thanks

Comment (czue) [reply docker/hq-compose-services.yml:48]: https://github.com/dimagi/commcare-hq/pull/29647

**Comment by orangejenny** [comment]
@czue Should that comment also be posted to the commcare developers forum?

**Comment by czue** [comment]
good call: https://forum.dimagi.com/t/changes-to-docker-elasticsearch-set-up/8177

**Comment by orangejenny** [comment]
@czue thanks!

**Pull request merged**